src/openapi_python_generator/language_converters/python/model_generator.py

```diff
@@ -136,9 +133,17 @@ def type_converter(  # noqa: C901
         else:
             converted_type = pre_type + "UUID" + post_type
             import_types = ["from uuid import UUID"]
-    elif schema.type == "string" and schema.schema_format == "date-time":
+    elif schema.type == "string" and schema.schema_format == "date-time" and common.get_use_orjson():
         converted_type = pre_type + "datetime" + post_type
         import_types = ["from datetime import datetime"]
+    elif schema.type == "string" and schema.schema_format == "date" and common.get_use_orjson():
+        converted_type = pre_type + "date" + post_type
+        import_types = ["from datetime import date"]
+    elif schema.type == "string" and schema.schema_format == "decimal" and common.get_use_orjson():
+        converted_type = pre_type + "Decimal" + post_type
+        import_types = ["from decimal import Decimal"]
+    elif schema.type == "string":
```

Comment:
This also captures unknown "schema_format" that would earlier result in "unknown type string" errors.